pkg/controller/servicebindingrequest/planner_test.go

```diff
@@ -21,13 +21,73 @@ func TestPlannerNew(t *testing.T) {
 	ns := "planner"
 	name := "service-binding-request"
 	resourceRef := "db-testing"
+
+	f := mocks.NewFake(t, ns)
+
+	// This test is here to stay, we may rename the function after
+	// deprecated fields are removed completely.
+	sbr := f.AddMockedServiceBindingRequestV1_1(name, resourceRef, "", deploymentsGVR)
+
+	f.AddMockedUnstructuredCSV("cluster-service-version")
+	f.AddMockedDatabaseCR(resourceRef)
+	f.AddMockedUnstructuredDatabaseCRD()
+
+	planner = NewPlanner(context.TODO(), f.FakeDynClient(), sbr)
+	require.NotNil(t, planner)
+
+	t.Run("searchCR", func(t *testing.T) {
+
+		for _, service := range *sbr.Spec.Services {
+			cr, err := planner.searchCR(ns, service)
+			require.NoError(t, err)
+			require.NotNil(t, cr)
+		}
+
+	})
+
+	t.Run("plan", func(t *testing.T) {
+		plan, err := planner.Plan()
+
+		require.NoError(t, err)
+		require.NotNil(t, plan)
+		require.NotEmpty(t, plan.RelatedResources)
+		require.Equal(t, ns, plan.Ns)
+		require.Equal(t, name, plan.Name)
+	})
+}
+
+func TestPlannerAnnotation(t *testing.T) {
+	ns := "planner"
+	name := "service-binding-request"
+	resourceRef := "db-testing"
+
+	f := mocks.NewFake(t, ns)
+	sbr := f.AddMockedServiceBindingRequestV1_1(name, resourceRef, "", deploymentsGVR)
```

Comment:
V1_1 ?

Reply:
Good question, the function name wasn't too explanatory, let me add a comment.
This function returns a ServiceBindingRequest with the newer API consisting of "applications" & "services".

Reply:
Added comment

```
// AddMockedServiceBindingRequestV1_1 add mocked object from ServiceBindingRequestMock.
// This function returns a ServiceBindingRequest with the newer API
// consisting of "applications" & "services".
```